### PR TITLE
:zap: Moved the initialisation to a dedicated function

### PR DESCRIPTION
**Pull request by ickule**

### CODE REVIEW

**Comment by ickule** [comment]
I would be interested to know what you think of the null checks on this PR.
It feels like a saner way but it feels like it would lead to a lot of copies of the same code throughout the program.
What do you think?

**Comment by ickule** [comment]
Moved unrelated code style changes to https://github.com/ickule/yagolic/pull/35 and https://github.com/ickule/yagolic/pull/37 to make reviewing easier.

File: include/board.h [hunk 1]
```diff
@@ -0,0 +1,10 @@
+#ifndef BOARD_H_INCLUDED
+#define BOARD_H_INCLUDED
+
+typedef struct {
+  int rows;
+  int cols;
+  int *cells;
+} Board;
```
Comment (kaoliiny) [inline include/board.h:8]: Rename to board_t

File: include/copy_board.h [hunk 2]
```diff
@@ -1,7 +1,11 @@
 #ifndef COPY_BOARD_H_INCLUDED
 #define COPY_BOARD_H_INCLUDED
 
-void copy_board(int rows, int cols, int *board, int *copy);
+#include <stdlib.h>
+
+#include "board.h"
+
+Board copy_board(Board board);
```
Comment (kaoliiny) [inline include/copy_board.h:8]: Board operations could be included into single h file

File: src/copy_board.c [hunk 3]
```diff
@@ -1,11 +1,16 @@
 #include "copy_board.h"
 
-void copy_board(int rows, int cols, int *board, int *copy)
+Board copy_board(Board board)
```
Comment (kaoliiny) [inline src/copy_board.c:3]: Board board board

File: src/yagolic.c [hunk 4]
```diff
@@ -1,39 +1,31 @@
...
       }
     }
     system("clear");
-    spread_life(board, rows, cols);
+    spread_life(board);
     i++;
   }
 
-  free(board);
+  free(board.cells);
```
Comment (kaoliiny) [inline src/yagolic.c:28]: Haven't we merge MR that deletes dynamic memory allocation?

**Comment by kaoliiny** [comment]
What the purpose to keep 7 h files for single board?

File: src/yagolic.c (hunk 4 above)
Comment (ickule) [reply src/yagolic.c:28]: No, I made changes that have not been approved yet.

**Comment by ickule** [comment]
That is a fair point, see https://github.com/ickule/yagolic/issues/39
I am not sure I will put all of them into a single file, but we can do some grouping.

**Comment by ickule** [comment]
I removed the null check part, I will handle it in the dedicated PR.

File: src/yagolic.c (hunk 4 above)
Comment (ickule) [reply src/yagolic.c:23]: I really think it makes more sense to allocate from heap here since we can't know the size of the board before execution.

Comment (kaoliiny) [reply src/yagolic.c:23]: and how heap allocation would help us?

**Review by kaoliiny — APPROVED** [review]

File: src/yagolic.c (hunk 4 above)
Comment (ickule) [reply src/yagolic.c:23]: The number of cells required is not know before compilation. 

So it is either :

1. We allocate from stack based on the user input and we end up with a Variable-length array. I don't really want to use something that is banned from the Linux kernel itself.
2. We allocate from stack with a big static size. This is wasteful of memory, and it needs dedicated checks on the user inputs. It feels dirty.
3. We allocate from heap. We use just the right amount of memory but we have to handle the memory ourselves and allocation/de-allocation will be slower.

Unless I missed something, option 3 sounds the best to me.
What do you think?

**Pull request merged**